ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/deneb/util/AttestationUtilDeneb.java

```diff
@@ -0,0 +1,92 @@
+  }
+
+  /**
+   * [IGNORE] attestation.data.slot is equal to or earlier than the current_slot (with a
+   * MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. attestation.data.slot <= current_slot (a
+   * client MAY queue future attestation for processing at the appropriate slot).
+   *
+   * <p>[IGNORE] the epoch of attestation.data.slot is either the current or previous epoch (with a
+   * MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. compute_epoch_at_slot(attestation.data.slot)
+   * in (get_previous_epoch(state), get_current_epoch(state))
+   */
+  @Override
+  public Optional<SlotInclusionGossipValidationResult> performSlotInclusionGossipValidation(
+      final Attestation attestation, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
+    final UInt64 attestationSlot = attestation.getData().getSlot();
+    final UInt64 attestationSlotTimeMillis =
+        secondsToMillis(miscHelpers.computeTimeAtSlot(genesisTime, attestationSlot));
+    if (isAttestationSlotAfterCurrentTime(attestationSlotTimeMillis, currentTimeMillis)) {
+      if (isFromFarFuture(attestation, genesisTime, currentTimeMillis)) {
+        return Optional.of(SlotInclusionGossipValidationResult.IGNORE);
+      }
+      return Optional.of(SlotInclusionGossipValidationResult.SAVE_FOR_FUTURE);
+    }
+    if (!isAttestationSlotInCurrentOrPreviousEpoch(
+        attestationSlot, genesisTime, currentTimeMillis)) {
+      return Optional.of(SlotInclusionGossipValidationResult.IGNORE);
+    }
+    return Optional.empty();
+  }
+
+  private boolean isAttestationSlotAfterCurrentTime(
+      final UInt64 attestationSlotTimeMillis, final UInt64 currentTimeMillis) {
+    return attestationSlotTimeMillis.isGreaterThan(
+        currentTimeMillis.plus(specConfig.getMaximumGossipClockDisparity()));
+  }
+
+  private boolean isAttestationSlotInCurrentOrPreviousEpoch(
+      final UInt64 attestationSlot, final UInt64 genesisTime, final UInt64 currentTimeMillis) {
+    final UInt64 currentSlot =
+        miscHelpers.computeSlotAtTime(genesisTime, millisToSeconds(currentTimeMillis));
+    final UInt64 currentEpoch = miscHelpers.computeEpochAtSlot(currentSlot);
+    final UInt64 previousEpoch = currentEpoch.minusMinZero(1);
+    final int slotDisparity = calculateMaximumGossipClockDisparityInSlots();
+    // min and max slot for the given attestation slot based on previous and current epoch with
+    // MAXIMUM_GOSSIP_CLOCK_DISPARITY
+    final UInt64 minSlot =
+        miscHelpers.computeStartSlotAtEpoch(previousEpoch).minusMinZero(slotDisparity);
+    final UInt64 maxSlot = miscHelpers.computeEndSlotAtEpoch(currentEpoch).plus(slotDisparity);
+
+    return attestationSlot.isGreaterThanOrEqualTo(minSlot)
+        && attestationSlot.isLessThanOrEqualTo(maxSlot);
+  }
+
+  private int calculateMaximumGossipClockDisparityInSlots() {
+    return (specConfig.getMaximumGossipClockDisparity() / specConfig.getMillisPerSlot()) + 1;
+  }
```

Comment:
likewise here, could definitely do some unit testing if we need to check any of this... it would show us our business logic is correct for `isAttestationSlotInCurrentOrPreviousEpoch` for example, and we could easily boundary test etc...